src/openrct2/actions/TrackPlaceAction.hpp

```diff
@@ -19,7 +19,31 @@
 #include "../world/Surface.h"
 #include "GameAction.h"
 
-DEFINE_GAME_ACTION(TrackPlaceAction, GAME_COMMAND_PLACE_TRACK, GameActionResult)
+class TrackPlaceActionResult final : public GameActionResult 
+{
+public:
+    TrackPlaceActionResult()
+    : GameActionResult(GA_ERROR::OK, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE)
+    {
+    }
+    TrackPlaceActionResult(GA_ERROR error)
+    : GameActionResult(error, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE)
+    {
+    }
+    TrackPlaceActionResult(GA_ERROR error, rct_string_id message)   
+    : GameActionResult(error, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, message)
+    {
+    }
+     TrackPlaceActionResult(GA_ERROR error, rct_string_id message, uint8_t* args)  
+    : GameActionResult(error, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, message, args)
+    {
+    }
+
+    uint8_t GroundFlag{ 0 };
```

Comment:
I think it would be better to call this GroundFlags

Reply:
Makes more sense for consistency I guess.